NitroxPatcher/Patches/Persistent/GameSettings_SerializeSettings_Patch.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Harmony;
+using NitroxClient.MonoBehaviours.Gui.Settings;
+using System;
+using System.Reflection;
+
+namespace NitroxPatcher.Patches.Persistent
+{
+    public class GameSettings_SerializeSettings_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(GameSettings);
+        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("SerializeSettings", BindingFlags.NonPublic | BindingFlags.Static);
+
+        public static void Postfix(GameSettings.ISerializer serializer)
+        {
+            SettingsManager.SetName(serializer.Serialize("Multiplayer/Name", SettingsManager.GetName()));
+            SettingsManager.SetKey_Chat(serializer.Serialize("Multiplayer/Chat/Primary", SettingsManager.GetKey_Chat()));
+            SettingsManager.SetColorR(serializer.Serialize("Multiplayer/colorRed", SettingsManager.GetColorR()));
```

Comment:
Doesn't the serializer support serializing a color directly?

Reply:
No. :(

Reply:
Also, is it really necessary to call the Set functions after serializing. As far as I know, they just return the value that was passed in.

Reply:
The serializer also read the settings for start.

Reply:
Yeah I just noticed by decompiling the code... The `Serialize` method deserializes, based on a `reading` boolean :/.